Created content for Extension API - Description, Features, and Considerations (#3)
infor-cloud/xtendm3: docs/documentation/api-specification/extension-api.md

**Category**: Generic  
---
## Description

> The Extension API provides information regarding the XtendM3 extension that is currently in use. This can be used as a validation in the development of the extension code as there can be other concurrent XtendM3 extensions activated.

Reviewer: ```suggestion
The Extension API provides information regarding the XtendM3 extension that is currently in use. This can be used as to retrieve metadata about the current extension and show to the user or use for logging purposes.
```

## Features
### Get creator
To retrieve the username of the extension creator. 

Example:
```groovy

> final ExtensionAPI extension

Reviewer: ```suggestion
final ExtensionAPI extension
```

> final ExtensionAPI extension

Reviewer: Examples should include a complete extension. 

```suggestion
public class SampleExtension extends ExtendM3Trigger {
  private final ExtensionAPI extension
  private final InteractiveAPI interactive
  
  public SampleExtension(ExtensionAPI extension, InteractiveAPI interactive) {
    this.extension = extension
    this.interactive = interactive
  }
  
  public void main() {
    String creator = extension.getCreator()
    interactive.showOkDialog("${creator} created this extension.")
  }
}
```
